orchagent/neighorch.cpp

```diff
@@ -433,6 +441,66 @@ bool NeighOrch::ifChangeInformNextHop(const string &alias, bool if_up)
     return rc;
 }
 
+void NeighOrch::updateNextHop(const BfdUpdate& update)
+{
+    SWSS_LOG_ENTER();
+    bool rc = true;
+
+    auto key = update.peer;
+    sai_bfd_session_state_t state = update.state;
+
+    size_t found_vrf = key.find(state_db_key_delimiter);
+    if (found_vrf == string::npos)
+    {
+        SWSS_LOG_INFO("Failed to parse key %s, no vrf is given", key.c_str());
+        return;
+    }
+
+    size_t found_ifname = key.find(state_db_key_delimiter, found_vrf + 1);
+    if (found_ifname == string::npos)
+    {
+        SWSS_LOG_INFO("Failed to parse key %s, no ifname is given", key.c_str());
+        return;
+    }
+
+    string vrf_name = key.substr(0, found_vrf);
+    string alias = key.substr(found_vrf + 1, found_ifname - found_vrf - 1);
+    IpAddress peer_address(key.substr(found_ifname + 1));
+
+    if (alias != "default" || vrf_name != "default")
+    {
+        return;
+    }
+
+    for (auto nhop = m_syncdNextHops.begin(); nhop != m_syncdNextHops.end(); ++nhop)
+    {
+        if (nhop->first.ip_address != peer_address)
+        {
+            continue;
+        }
+
+        if (state == SAI_BFD_SESSION_STATE_UP)
+        {
+            SWSS_LOG_INFO("updateNextHop get BFD session UP event, key %s", key.c_str());
+            rc = clearNextHopFlag(nhop->first, NHFLAGS_IFDOWN);
```

Comment:
So this is going to update all routes that has this nexthop, irrespective of whether BFD is set to true for such route or not, correct?

Reply:
Yes. From BFD SESSION itself (or BFD state event), it cannot tell if it is created by StaticRouteBfd or not.

Reply:
@abdosi for viz